packages/syft/src/syft/serde/lib_service_registry.py

```diff
@@ -129,65 +137,76 @@ def init_child(
+            return None
+
+        if inspect.ismodule(child_obj):
+            ## TODO, we could register modules and functions in 2 ways:
+            # A) as numpy.float32 (what we are doing now)
+            # B) as numpy.core.float32 (currently not supported)
+            # only allow submodules
+            return CMPModule(
                 child_path,
                 permissions=self.permissions,
                 obj=child_obj,
                 absolute_path=absolute_path,
             )  # type: ignore
-        elif inspect.ismodule(child_obj) and CMPBase.is_submodule(
-            parent_obj, child_obj
-        ):
-            ## TODO, we could register modules and functions in 2 ways:
-            # A) as numpy.float32 (what we are doing now)
-            # B) as numpy.core.float32 (currently not supported)
-            # only allow submodules
 
-            return CMPModule(
+        # Here we have our own isfunction as there are multiple callable objects worth considering
+        if CMPBase.isfunction(child_obj):
+            return CMPFunction(
                 child_path,
                 permissions=self.permissions,
                 obj=child_obj,
                 absolute_path=absolute_path,
             )  # type: ignore
-        elif inspect.isclass(child_obj) and parent_is_parent_module:
+
+        if inspect.isclass(child_obj):
             return CMPClass(
                 child_path,
                 permissions=self.permissions,
                 obj=child_obj,
                 absolute_path=absolute_path,
             )  # type: ignore
-        else:
-            return None
+
+        # default case if we didnt cover it
+        # currently used for objects
+        return CMPBase(
+            child_path,
+            permissions=self.permissions,
+            obj=child_obj,
+            absolute_path=absolute_path,
+        )
 
     @staticmethod
-    def is_submodule(parent: type, child: type) -> bool:
+    def check_package_membership(parent_obj: Any, child_obj: Any) -> bool:
+        # we are wrapping this as some objects might not have some of the dunder methods
+        # TODO: check if that is truly necessary
         try:
-            if "." not in child.__package__:
+            # if we find the same name we should avoid a circular import (probably obsolete)
+            if child_obj.__name__ == parent_obj.__name__:
                 return False
-            else:
-                child_parent_module = child.__package__.rsplit(".", 1)[0]
-                if parent.__package__ == child_parent_module:
-                    return True
-                else:
-                    return False
-        except Exception:  # nosec
-            pass
-        return False
 
-    @staticmethod
-    def parent_is_parent_module(parent_obj: Any, child_obj: Any) -> Optional[str]:
-        try:
+            # if the name of the parent can be found at the start of the name of the child we are good
```

Comment:
does this assume both child and parent are modules? I could see how this would easily give wrong results if they wouldnt

Reply:
I suspected that was the case for why initially we wrapped that function in a try expect. Of course, if the parent is not a submodule it is fine to return False (maybe there should be problems with static methods of some classes, but at the same time we need some rigorous testing as there are still corner cases that we miss, I assumed we would do that at the end of the quarter